hdf/src/hdf.h

```diff
@@ -149,4 +189,37 @@ typedef intn (*hdf_termfunc_t)(void); /* termination function typedef */
 /* This is also defined in fmpio.h */
 #define MP_PAGEALL 0x01 /* page the whole file i.e. no limit on 'maxcache'*/
 
+/**************************************************************************
+ *  Memory and string functions
+ **************************************************************************/
+
+/* DO NOT USE THESE MACROS */
+
+/* These will be removed from a future version of the library and are
+ * only kept here to avoid breakage in programs that unwisely used
+ * them.
+ */
+
+#define HDmalloc(s)     malloc(s)
+#define HDcalloc(a, b)  calloc(a, b)
+#define HDfree(p)       free(p)
+#define HDrealloc(p, s) realloc(p, s)
+
+/* Macro to free space and clear pointer to NULL */
+#define HDfreenclear(p)                                                                                      \
+    {                                                                                                        \
+        free(p);                                                                                             \
+        (p) = NULL;                                                                                          \
+    }
+
+#define HDstrcat(s1, s2)     (strcat((s1), (s2)))
+#define HDstrcmp(s, t)       (strcmp((s), (t)))
+#define HDstrcpy(s, d)       (strcpy((s), (d)))
+#define HDstrlen(s)          (strlen((const char *)(s)))
+#define HDstrncmp(s1, s2, n) (strncmp((s1), (s2), (n)))
+#define HDstrncpy(s1, s2, n) (strncpy((s1), (s2), (n)))
+#define HDstrchr(s, c)       (strchr((s), (c)))
+#define HDstrrchr(s, c)      (strrchr((s), (c)))
+#define HDstrtol(s, e, b)    (strtol((s), (e), (b)))
```

Comment:
I'm aware of code that uses these API calls so they need to be public until we remove them